src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -195,9 +195,8 @@ private IServiceCollection BuildHostingServices()
             services.AddTransient<IHttpContextFactory, HttpContextFactory>();
             services.AddOptions();
 
-            var diagnosticSource = new DiagnosticListener("Microsoft.AspNetCore");
-            services.AddSingleton<DiagnosticSource>(diagnosticSource);
-            services.AddSingleton<DiagnosticListener>(diagnosticSource);
+            services.AddSingleton<DiagnosticListener>(provider => new DiagnosticListener("Microsoft.AspNetCore"));
+            services.AddSingleton<DiagnosticSource>(provider => provider.GetService<DiagnosticListener>());
```

Comment:
We don't have multi interface registration and there is no need to have this as an instance as far as I understand.